dmn/src/main/java/io/camunda/zeebe/dmn/impl/EvaluatedDmnScalaDecision.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.dmn.impl;
+
+import static java.util.Map.entry;
+
+import io.camunda.zeebe.dmn.DecisionType;
+import io.camunda.zeebe.dmn.EvaluatedDecision;
+import io.camunda.zeebe.dmn.EvaluatedInput;
+import io.camunda.zeebe.dmn.MatchedRule;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.agrona.DirectBuffer;
+import org.camunda.dmn.Audit.AuditLogEntry;
+import org.camunda.dmn.Audit.DecisionTableEvaluationResult;
+import org.camunda.dmn.parser.ParsedContext;
+import org.camunda.dmn.parser.ParsedDecisionLogic;
+import org.camunda.dmn.parser.ParsedDecisionTable;
+import org.camunda.dmn.parser.ParsedInvocation;
+import org.camunda.dmn.parser.ParsedList;
+import org.camunda.dmn.parser.ParsedLiteralExpression;
+import org.camunda.dmn.parser.ParsedRelation;
+import org.camunda.dmn.parser.ParsedRule;
+import org.camunda.feel.syntaxtree.Val;
+
+public record EvaluatedDmnScalaDecision(
+    String decisionId,
+    String decisionName,
+    DecisionType decisionType,
+    DirectBuffer decisionOutput,
+    List<EvaluatedInput> evaluatedInputs,
+    List<MatchedRule> matchedRules)
+    implements EvaluatedDecision {
+
+  private static final Map<Class<? extends ParsedDecisionLogic>, DecisionType>
+      DECISION_TYPE_MAPPING =
+          Map.ofEntries(
+              entry(ParsedDecisionTable.class, DecisionType.DECISION_TABLE),
+              entry(ParsedLiteralExpression.class, DecisionType.LITERAL_EXPRESSION),
+              entry(ParsedContext.class, DecisionType.CONTEXT),
+              entry(ParsedList.class, DecisionType.LIST),
+              entry(ParsedRelation.class, DecisionType.RELATION),
+              entry(ParsedInvocation.class, DecisionType.INVOCATION));
+
+  public static EvaluatedDmnScalaDecision of(
+      final AuditLogEntry auditLogEntry, final Function<Val, DirectBuffer> converter) {
+    final DecisionType decisionType = getDecisionType(auditLogEntry.decisionLogic());
+    final var evaluationResult = auditLogEntry.result();
+    final var decisionOutput = converter.apply(evaluationResult.result());
+
+    final var evaluatedInputs = new ArrayList<EvaluatedInput>();
+    final var matchedRules = new ArrayList<MatchedRule>();
+
+    if (evaluationResult instanceof DecisionTableEvaluationResult decisionTableResult) {
```

Comment:
❓  What if it's not an instance of? Do we want to allow creating instances of EvaluatedDmnScalaDecision that don't contain any matched rules and evaluated inputs? It's probably controlled from outside, but we should consider whether we want to be a bit more defensive here. What do you think?

Reply:
The DMN engine can evaluate more than just decision tables. Officially, we support also literal expressions because of Camunda Platform and the docs. I don't see a need to limit us here. 

If the decision is not a table then it just contains the output. Should be fine for now.

Reply:
That makes sense actually. I'm still getting used to only tables have inputs, but it makes sense.